File: submit/src/main/java/material/com/submit/api/SubmitService.java
package material.com.submit.api;

import material.com.submit.model.SubmitResult;
import io.reactivex.Observable;
import retrofit2.http.Field;
import retrofit2.http.POST;

/**
 * Created by cangwang on 2017/3/30.
 */

public interface SubmitService {
    @POST("add2gank")
    Observable<SubmitResult> submitGank(@Field("url") String url
            ,@Field("desc") String desc
            ,@Field("who")String who
            ,@Field("type") String type
            ,@Field("debug")boolean debug);
}

File: submit/src/main/java/material/com/submit/api/SubmitService.kt
package material.com.submit.api

import material.com.submit.model.SubmitResult
import io.reactivex.Observable
import retrofit2.http.Field
import retrofit2.http.POST

/**
 * Created by cangwang on 2017/3/30.
 */

interface SubmitService {
    @POST("add2gank")
    fun submitGank(@Field("url") url: String, @Field("desc") desc: String, @Field("who") who: String, @Field("type") type: String, @Field("debug") debug: Boolean): Observable<SubmitResult>
}
